File: batoid/coordtransform2.py
from . import _batoid

class CoordTransform2:
    def __init__(self, fromSys, toSys):
        self.fromSys = fromSys
        self.toSys = toSys
        self._coordTransform = _batoid.CPPCoordTransform2(
            fromSys._coordSys,
            toSys._coordSys
        )

    def applyForwardInPlace(self, r):
        self._coordTransform.applyForwardInPlace(r._rv)
        r.coordSys = self.toSys

    def applyReverseInPlace(self, r):
        self._coordTransform.applyReverseInPlace(r._rv)
        r.coordSys = self.fromSys

File: pysrc/coordtransform2.cpp
#include "coordtransform2.h"
#include <pybind11/pybind11.h>

namespace py = pybind11;

namespace batoid {
    void pyExportCoordTransform2(py::module& m) {
        py::class_<CoordTransform2, std::shared_ptr<CoordTransform2>>(m, "CPPCoordTransform2")
            .def(py::init<const CoordSys&, const CoordSys&>())
            .def("applyForwardInPlace", (void (CoordTransform2::*)(RayVector2&) const) &CoordTransform2::applyForwardInPlace)
            .def("applyReverseInPlace", (void (CoordTransform2::*)(RayVector2&) const) &CoordTransform2::applyReverseInPlace);
    }
}
